src/main/java/edu/cornell/kfs/sys/service/impl/CUMarshalServiceImpl.java

```diff
@@ -141,6 +156,22 @@ public <T> T unmarshalStream(InputStream inputStream, Class<T> clazz)
         Unmarshaller unmarshaller = createUnmarshaller(clazz, Optional.empty());
         return unmarshalStream(inputStream, unmarshaller);
     }
+    
+    @Override
+    public <T> T unmarshalStreamIgnoreDtd(InputStream inputStream, Class<T> clazz)
+            throws JAXBException, IOException, XMLStreamException {
+        JAXBContext jc = JAXBContext.newInstance(clazz);
+
+        XMLInputFactory inputFactory = XMLInputFactory.newFactory();
+        inputFactory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+        XMLStreamReader streamReader = inputFactory.createXMLStreamReader(inputStream);
```

Comment:
I would suggest specifying the charset when creating the stream reader (or create and use an intermediate `InputStreamReader` that specifies the charset). Sorry for not mentioning this earlier.

Reply:
inputFactory.createXMLStreamReader doesn't have an arg option for a character set, but it does have one that takes an encoding string, so I used the following and seems to work in a unit test, I will do an application test before I open this for review again
`streamReader = inputFactory.createXMLStreamReader(wrappedStream, StandardCharsets.UTF_8.name());`